R/formatter.r

```diff
@@ -767,49 +767,32 @@ number_bytes_format <- function(symbol = "auto", units = "binary", ...) {
 #' @export
 #' @rdname number_bytes_format
 number_bytes <- function(x, symbol = "auto", units = c("binary", "si"), ...) {
+  units <- match.arg(units, c("binary", "si"))
+
   symbols <- c(
-    "auto",
-    "b", "Kb", "Mb", "Gb", "Tb", "Pb", "Eb", "Zb", "Yb",
-    "B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB",
-    "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"
+    "auto", "B",
+    switch(units,
+      si     = c( "kB",  "MB",  "GB",  "TB",  "PB",  "EB",  "ZB",  "YB"),
```

Comment:
I wonder if it would be better to generate these from a data structure like `c(k = 3, M = 6, G = 9, ...)`, which could then be reused when automatically finding the symbol.

Reply:
@hadley yes I think that would be neater. I was thinking maybe it would make sense to just include a metric prefix table as package data and use it here? E.g. from [here](https://en.wikipedia.org/wiki/Metric_prefix)? Could include powers, symbols and text names for prefixes. It could also be used with `number_si()` and in other places, too, I'm sure.

Reply:
I can open another PR for that if that sounds good.

Reply:
@hadley I added the SI prefix table in #177, and it's working; but now I'm thinking maybe this got a bit over-engineered... Maybe it would be simpler to just include a vector like you suggested. What do you think?

Reply:
Yeah, I think a vector would be better, although embedding the data frame directly isn't too bad if you think the other columns would be useful in the future.

Reply:
I think having a single place for all the SI prefixes would be useful, but perhaps not so much the columns other than power and symbol. So I went with a middle-ground and included a named vector outside the function.